src/osm_callbacks.h

```diff
@@ -0,0 +1,23 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+#ifndef TIMESCALEDB_OSM_CALLBACKS_H
+#define TIMESCALEDB_OSM_CALLBACKS_H
+
+#include <postgres.h>
+#include <catalog/objectaddress.h>
+
+typedef int (*chunk_insert_check_hook_type)(Oid ht_oid, int64 range_start, int64 range_end);
+typedef void (*hypertable_drop_hook_type)(const char *schema_name, const char *table_name);
+
+typedef struct
+{
+	chunk_insert_check_hook_type chunk_insert_check_hook;
+	hypertable_drop_hook_type hypertable_drop_hook;
+} OsmCallbacks;
```

Comment:
I think its a bit unfortunate that the API defines the number of calls directly - if you want to add a new method you will need a second version....

how about approaching differently:
* add just a single method
* an opcode why the fook was called
* a context object with the currently known infos (table cat+name)

this way later on
* adding a new call is quite simple - just add a new enum value
* passing new infos is trivial since the old context is the prefix of the new
* no need to recompile old hooks when rolling out a new version of the...it will work without issues